src/macros.rs

```diff
@@ -53,6 +53,44 @@ macro_rules! redis_command {
     }};
 }
 
+#[cfg(feature = "experimental-api")]
+#[macro_export]
+macro_rules! redis_event_handler {
+    (
+        $ctx: expr,
+        $event_type: expr,
+        $event_handler: expr
+    ) => {{
+        extern "C" fn handle_event(
+            ctx: *mut raw::RedisModuleCtx,
+            event_type: c_int,
+            event: *const c_char,
+            key: *mut raw::RedisModuleString
+        ) -> c_int {
+            let context = Context::new(ctx);
+
+            let redis_key = RedisString::from_ptr(key).unwrap();
```

Comment:
This (and the `event_str.to_str().unwrap()` below) will cause a panic if the strings are not valid UTF-8, which IIRC Redis doesn't guarantee. I believe this would hardly be the only place in the crate where this issue may happen. @gavrie @gkorland Do we have any policy about this? (Asking in general, not for this PR specifically.)

Reply:
@slavak This is indeed a problematic assumption, and I think we make it in several places as you said. Not a problem specific to this PR, we should think about what we want to do here in general.